controller/destination/listener.go

```diff
@@ -152,14 +152,13 @@ func (l *endpointListener) toWeightedAddrSet(addresses []*updateAddress) *pb.Wei
 }
 
 func (l *endpointListener) toWeightedAddr(address *updateAddress) *pb.WeightedAddr {
-	metricLabelsForPod := pkgK8s.GetOwnerLabels(address.pod.ObjectMeta)
-	metricLabelsForPod["pod"] = address.pod.Name
+	labels, tlsIdentity := l.getAddrMetadata(address.pod)
```

Comment:
This behaves slightly differently than what's currently on master. When formulating the set of labels to send back to the proxy, we used to rely exclusively on the pod's injected `linkerd.io/*` labels. In this branch I'm still reading the controller namespace and pod template hash from the pod's labels, but I'm relying on the kubernetes API to generate the owner label (deployment, replicationcontroller, etc.). This is more consistent with how we construct TLS identities, and it allows us to record outbound stats to un-injected pods.

For instance, if I inject only the web deployment from the [emojivoto demo](https://github.com/runconduit/conduit-examples/tree/master/emojivoto) using the code that's currently on master, then the deploy stats look like this:

```
$ linkerd stat -n emojivoto deploy
NAME       MESHED   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99   TLS
emoji         0/1         -        -             -             -             -     -
vote-bot      0/1         -        -             -             -             -     -
voting        0/1         -        -             -             -             -     -
web           1/1    89.57%   1.9rps           3ms          10ms          28ms    0%
```

If I add a `--from` flag, no deploys are returned:

```
$ linkerd stat -n emojivoto deploy --from deploy
No traffic found.
```

Whereas if I recreate the same setup on this branch and add a `--from` flag, I get back data:

````
$ bin/linkerd stat -n emojivoto deploy --from deploy
NAME     MESHED   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99   TLS
emoji       0/1   100.00%   2.0rps           1ms           1ms           2ms    0%
voting      0/1    81.36%   1.0rps           1ms           1ms           1ms    0%
```